### PR TITLE
[Merged by Bors] - feat(Data/Set/Intervals/Disjoint): i[Inter|Union]_Ii[c|o]...

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
This prepares removing some lemmas from `Probability/Kernel/CondCdf`. [Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/good.20place.20for.20.60iUnion_Iic_rat_coe.60)

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 1]
```diff
@@ -253,4 +255,26 @@ theorem iUnion_Iic_eq_Iic_iSup {R : Type*} [CompleteLinearOrder R] {f : ι → R
   @iUnion_Ici_eq_Ici_iInf ι (OrderDual R) _ f has_greatest_elem
 #align Union_Iic_eq_Iic_supr iUnion_Iic_eq_Iic_iSup
 
+theorem iUnion_Iio_of_not_bddAbove_range (hf : ¬ BddAbove (range f)) :
```
Comment (urkud) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:258]: Please add order-dual versions (probably using the `OrderDual` trick).

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 2]
```diff
@@ -253,4 +255,26 @@ theorem iUnion_Iic_eq_Iic_iSup {R : Type*} [CompleteLinearOrder R] {f : ι → R
...
+    ⋃ i, Iio (f i) = univ := by
+  simpa [not_bddAbove_iff, Set.eq_univ_iff_forall] using hf
+
+theorem iUnion_Iic_of_not_bddAbove_range (hf : ¬ BddAbove (range f)) :
+    ⋃ i, Iic (f i) = univ := by
+  rw [← Set.univ_subset_iff]
```
Comment (urkud) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:264]: Probably, you can golf using `Set.eq_univ_of_subset`.

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 3]
```diff
@@ -253,4 +255,26 @@ theorem iUnion_Iic_eq_Iic_iSup {R : Type*} [CompleteLinearOrder R] {f : ι → R
   @iUnion_Ici_eq_Ici_iInf ι (OrderDual R) _ f has_greatest_elem
 #align Union_Iic_eq_Iic_supr iUnion_Iic_eq_Iic_iSup
 
+theorem iUnion_Iio_of_not_bddAbove_range (hf : ¬ BddAbove (range f)) :
+    ⋃ i, Iio (f i) = univ := by
+  simpa [not_bddAbove_iff, Set.eq_univ_iff_forall] using hf
```
Comment (urkud) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:260]: `simpa` says that it can be an `iff` lemma.

Comment (mo271) [reply Mathlib/Data/Set/Intervals/Disjoint.lean]: Made it an `iff` lemma. 
Is the name good like this?

File: Mathlib/Data/Set/Intervals/Disjoint.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/Data/Set/Intervals/Disjoint.lean]: I thought the version below, now renamed  `iUnion_Iio_eq_univ_iff` and `iInter_Iio_eq_empty_iff` are duals of each other. Do you mean some other sort of dual statement? If yes, which one?

Comment (mo271) [reply Mathlib/Data/Set/Intervals/Disjoint.lean]: ping @urkud

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 4]
```diff
@@ -257,4 +259,21 @@ theorem iUnion_Iic_eq_Iic_iSup {R : Type*} [CompleteLinearOrder R] {f : ι → R
...
 
+theorem iUnion_Iio_eq_univ_iff : ⋃ i, Iio (f i) = univ ↔ (¬ BddAbove (range f)) := by
+  simp [not_bddAbove_iff, Set.eq_univ_iff_forall]
+
+theorem iUnion_Iic_of_not_bddAbove_range (hf : ¬ BddAbove (range f)) : ⋃ i, Iic (f i) = univ := by
+  refine  Set.eq_univ_of_subset ?_ (iUnion_Iio_eq_univ_iff.mpr hf)
+  gcongr
+  exact Iio_subset_Iic_self
+
+theorem iInter_Iic_eq_iff : ⋂ i, Iic (f i) = ∅ ↔ ¬ BddBelow (range f) := by
```
Comment (RemyDegenne) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:270]:
```suggestion
theorem iInter_Iic_eq_empty_iff : ⋂ i, Iic (f i) = ∅ ↔ ¬ BddBelow (range f) := by
```

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 5]
```diff
@@ -257,4 +259,21 @@ theorem iUnion_Iic_eq_Iic_iSup {R : Type*} [CompleteLinearOrder R] {f : ι → R
...
+  simp [not_bddBelow_iff, Set.eq_empty_iff_forall_not_mem]
+
+theorem iInter_Iio_eq_empty_iff (hf : ¬ BddBelow (range f)) : ⋂ i, Iio (f i) = ∅ := by
```
Comment (RemyDegenne) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:273]:
```suggestion
theorem iInter_Iio_of_not_bddBelow_range (hf : ¬ BddBelow (range f)) : ⋂ i, Iio (f i) = ∅ := by
```

**Review by RemyDegenne — APPROVED** [review]
I just have comments about the names, to make them match the other lemmas you introduce. Apart from that, LGTM.
bors d+

**Comment by mo271** [comment]
will fix the build...

**Comment by mo271** [comment]
fixed the name and rebased

**Pull request closed without merging**